scripts/python/src/fodt/split_git_commit.py

```diff
@@ -0,0 +1,242 @@
+        if self.repo.is_dirty(untracked_files=True):
+            self.check_for_uncommitted_changes()
+        last_commit_files = self.get_last_commit_changed_files()
+        fodt_file = self.validate_and_get_fodt_file(last_commit_files)
+        self.create_backup_branch()
+        self.save_commit_message()
+        self.split_commit_two_parts(fodt_file)
+        self.verify_no_difference_with_backup()
+
+    def split_commit_two_parts(self, fodt_file: str) -> None:
+        """Split the commit into two parts for style and content changes."""
+        # Extract current and previous versions of the file
+        current_file, previous_file = self.extract_file_versions(fodt_file)
+
+        # Identify style and content changes
+        style_change_content = self.get_style_change(current_file, previous_file)
+        content_change_content = self.get_content_change(current_file)
+        self.delete_current_commit()
+
+        # Create commits for style and content changes
+        self.commit_changes(
+            fodt_file,
+            style_change_content,
+            f"{self.saved_commit_message} - Style changes"
+        )
+        self.commit_changes(
+            fodt_file,
+            content_change_content,
+            f"{self.saved_commit_message} - Content changes"
+        )
+
+    def validate_and_get_fodt_file(self, last_commit_files: list[str]) -> str:
+        """Validate the last commit and return the .fodt filename."""
+        if len(last_commit_files) == 1 and last_commit_files[0].endswith('.fodt'):
+            return last_commit_files[0]
+        else:
+            print("Error: No .fodt file specified and last commit does not "
+                  "contain exactly one .fodt file.")
+            sys.exit(1)
+
+    def verify_no_difference_with_backup(self) -> None:
+        """Verify that there is no difference between the current branch and the backup branch."""
+        diff = self.repo.git.diff(self.backup_branch_name, 'HEAD')
+        if diff:
+            logging.error(
+                "Error: There is a difference between the backup branch and the current state."
+            )
+        else:
+            logging.info(
+                "Verification successful: No difference between the backup branch and "
+                "the current state."
+            )
+
+# fodt-split-commit
+# -----------------
+# SHELL USAGE
+# -----------
+#
+# fodt-split-commit
+#
+# DESCRIPTION
+# -----------
+# Split the last committed file in a Git repository into separate commits for style and
+#  content changes. To simplify complexities with stashing of files, the last commit
+#  should contain changes to a single .fodt file. If the last commit contains
+#  changes to multiple files en error will be raised. The user is then required
+#  to manually separate the changes into separate commits before running this script.
+#  If the last commit contain change to a file that is not a .fodt file, an error
```

Comment:
contain*s* change*s*

Reply:
Thanks! I have updated the PR to fix the spelling